esphome/components/esp32/gpio.py

```diff
@@ -69,6 +69,8 @@ def _translate_pin(value):
         )
     if isinstance(value, int):
         return value
+    if isinstance(value, list):
+        raise cv.Invalid("A pin schema can't be a list.")
```

Comment:
At this point the value should only be a string. Ditto for the other cases.
```suggestion
    if not isinstance(value, str):
        raise cv.Invalid(f"Invalid pin number: {value}")
```